sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/FileFormat.scala

```diff
@@ -152,6 +152,11 @@ trait FileFormat {
     }
   }
 
+  /**
+   * Returns whether this format supports the given [[DataType]] in read/write path.
+   * By default all data types are supported.
+   */
+  def supportDataType(dataType: DataType, isReadPath: Boolean): Boolean = true
```

Comment:
Technically this is an internal API but you keep the compatibility here in practice. I think I already see we are concerned about Avro, right? I will doubt if it's a good idea to expose this in this trait.

Reply:
This is the only way to allow avro to define its supported types. BTW the default true value here is good for compatibility: if a file source doesn't know this API, it doesn't need to implement it and the behavior is unchanged, which is, no check applied.

Reply:
I meant to say it's not free to remove or change the signature later once we happen to add it. Do we plan to refactor or remove this out in the near future? If so I am okay.

Reply:
the entire FileFormat will be migrated to data source v2 in the future. The FileFormat will be still there for backward compatibility, but I don't think we will update it frequently.